clippy_lints/src/cargo/mod.rs

```diff
@@ -155,12 +155,25 @@ declare_clippy_lint! {
     /// [As the edition guide says](https://rust-lang-nursery.github.io/edition-guide/rust-2018/cargo-and-crates-io/crates-io-disallows-wildcard-dependencies.html),
     /// it is highly unlikely that you work with any possible version of your dependency,
     /// and wildcard dependencies would cause unnecessary breakage in the ecosystem.
+    /// Instead, you must actually specify a specific version or range of versions, using one of the `semver` crate’s various options: `^`, `~`, or `=`.
```

Comment:
Having this description in a section called "Why is this bad?" is quite strange. This lint is also simple enough that the example shows this better.